src/main/scala/org/bblfsh/client/v2/JNode.scala

```diff
@@ -1,15 +1,60 @@
 package org.bblfsh.client.v2
 
+import scala.collection.mutable
+
 /** UAST nodes representation on the JVM side.
  *
  *  Mirrors https://godoc.org/github.com/bblfsh/sdk/uast/nodes
  */
-sealed abstract class JNode
-case class JString(s: String) extends JNode
-case class JDouble(num: Double) extends JNode
-case class JLong(num: Long) extends JNode
-case class JInt(num: BigInt) extends JNode
-case class JBool(value: Boolean) extends JNode
-case class JObject(obj: List[JField]) extends JNode
-case class JArray(arr: List[JNode]) extends JNode
+sealed abstract class JNode {
+  def children: Seq[JNode] = this match {
+    case JObject(l) => l map (_._2)
+    case JArray(l) => l
+    case _ => Seq()
+  }
+
+  def size: Int = this match {
```

Comment:
I don't know what's idiomatic for Scala, but it seems slightly odd to do this as a dispatch on type in the base class rather than as an override in the subclasses with a default here. If the subclasses don't add any behaviour, do they need to be related at all?

Reply:
In general, yes, that is true, but in our particular case - this is just a convenience, in order to avoid having this exact logic on JNI side.